apps/backend/src/modules/users/user.service.ts

```diff
@@ -101,9 +133,16 @@ class UserService implements Service {
 			});
 		}
 
-		const updatedUser = await this.userRepository.update(id, updateData);
+		try {
+			const updatedUser = await this.userRepository.update(id, updateData);
 
-		return updatedUser ? updatedUser.toObject() : null;
+			return updatedUser ? updatedUser.toObject() : null;
+		} catch {
+			throw new HTTPError({
+				message: UserValidationMessage.FAILED_TO_UPDATE_PROFILE,
+				status: HTTPCode.INTERNAL_SERVER_ERROR,
+			});
+		}
```

Comment:
Move this part to a separate function so that you can call it in case `!payload.password?.trim()` and in case `!payload.password?.trim()`